src/api/addSecurityQuestion.js

```diff
@@ -43,17 +41,39 @@ router.post(
       return;
     } else {
       let authenticated = false;
-      if (await bcrypt.compare(req.body.password, user.password)) {
+      if (
+        !!req.body.password &&
+        (await bcrypt.compare(req.body.password, user.password))
+      ) {
         // hash matches! sign a JWT with an expiration 1 day in the future and send back to the user
         authenticated = true;
       }
       if (!authenticated) {
         return sendResponse(res, 400, "Incorrect Password");
       }
-      await User.updateOne(
+      const securityQuestionsResponse = await getSecurityQuestions();
+      let question;
```

Comment:
remove these and define consts later to clean up the code
let question;
const { answer } = req.body;